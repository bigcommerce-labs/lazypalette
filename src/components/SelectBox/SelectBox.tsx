import React from 'react';

import StyledDiv from './styles/StyledDiv';
import StyledLabel from './styles/StyledLabel';
import StyleOption from './styles/StyledOption';
import StyleSelect from './styles/StyledSelect';

interface SelectBoxProps {
    label?: string;
    options: string[];
}

class SelectBox extends React.Component<SelectBoxProps, {}> {
    static defaultProps = {
        label: undefined,
        options: undefined,
    };

    render() {
        const { label, options } = this.props;
        const StyleOptions = options.map((option: string) => (
            <StyleOption key={window.btoa(option)}>
                { option }
            </StyleOption>
        ));

        return (
            <StyledDiv>
                <StyledLabel>
                    { label }
                </StyledLabel>
                <StyleSelect>
                    { StyleOptions }
                </StyleSelect>
            </StyledDiv>
        );
    }
}

export default SelectBox;
