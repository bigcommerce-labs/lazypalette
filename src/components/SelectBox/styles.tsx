import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Label = styled.label`
  padding: .3125rem;
`;

export const Select = styled.select`
  height: 2.1875rem;
  width: 100%;
  font-size: .9375rem;
  color: ${({ theme }) => theme.colors.primaryText};
  padding: .375rem 1.5625rem .375rem .5rem;
  background-color: ${({ theme }) => theme.colors.background};
  border: 1px solid ${({ theme }) => theme.colors.stroke};
  cursor: pointer;
`;
