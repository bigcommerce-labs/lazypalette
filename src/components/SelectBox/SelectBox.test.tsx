import { shallow } from 'enzyme';
import React from 'react';

import SelectBox from './SelectBox';

it('renders', () => {
  const selectBox = shallow(
    <SelectBox label="Label" options={[]} />
  );
  expect(selectBox).toMatchSnapshot();
});
