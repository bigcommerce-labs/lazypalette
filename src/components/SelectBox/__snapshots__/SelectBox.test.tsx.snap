// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.div>
  <styled.label>
    Label
  </styled.label>
  <styled.select />
</styled.div>
`;
